dev/doc/README.md

```diff
@@ -0,0 +1,90 @@
+# Beginner's guide
+
+## Getting dependencies
+
+Assuming one is running Ubuntu (if not, replace `apt` with the package manager of choice)
+
+```
+$ sudo apt-get install make opam git
+$ opam init --comp 4.02.3
+# Then follow the advice displayed at the end as how to update your
+  ~/.bashrc and ~/.ocamlinit files.
+
+$ source ~/.bashrc
+
+# needed if you want to build "coqtop" target
+$ opam install camlp5
+
+# needed if you want to build "coqide" target
+$ sudo apt-get install liblablgtksourceview2-ocaml-dev \
+  libgtk2.0-dev libgtksourceview2.0-dev
+$ opam install lablgtk
+
+# needed if you want to build "doc" target
+$ sudo apt-get install texlive-latex-recommended \
+  texlive-fonts-extra texlive-math-extra \
+  hevea texlive-latex-extra latex-xcolor
+```
+
+## Building `coqtop`
+The general workflow is to first setup a development environment with
+the correct `configure` settings, then hacking on Coq, make-ing, and testing.
+
+
+This document will use `$JOBS` to refer to the number of parallel jobs one
+is willing to have with `make`.
+
+
+```
+$ git clone git clone https://github.com/coq/coq.git
+$ cd coq
+$ ./configure -profile devel
+$ make -j $JOBS # Make once for `merlin`(autocompletion tool)
+
+<hack>
+$ make -j $JOBS states # builds just enough to run coqtop
+$ bin/coqtop -compile <test_file_name.v>
+<goto hack until stuff works>
+
+$ make -C test-suite -j $JOBS # run test suite
+```
+
+
+To run the test suite, run `make` from the `test-suite` directory.
+This is explained in greater detail in [`/test-suite/README.md`](/test-suite/README.md)
```

Comment:
I recently discovered that relative links were better, see #7800. So you can use `../../test-suite/README.md` instead.